appium/webdriver/webdriver.py

```diff
@@ -581,8 +582,23 @@ def lock(self, seconds):
             'seconds': seconds,
         }
         self.execute(Command.LOCK, data)
+        if self.desired_capabilities.get('platformName', '').lower() == 'android':
+            WebDriverWait(self, 10).until(lambda d: d.is_locked())
```

Comment:
i agree that the server should be responsible for only returning when we're in a known locked state